test/Microsoft.SqlTools.ServiceLayer.Test/QueryExecution/Execution/ServiceIntegrationTests.cs

```diff
@@ -49,14 +51,23 @@ public class ServiceIntegrationTests
             // ... A completion event should have been fired with empty results
             // ... A batch completion event should have been fired with empty results
             // ... A result set completion event should not have been fired
-            VerifyQueryExecuteCallCount(requestContext, Times.Once(), Times.Once(), Times.Once(), Times.Never(), Times.Never());
+            VerifyQueryExecuteCallCount(requestContext, Times.Once(), Times.Once(), Times.Once(), Times.Once(), Times.Never(), Times.Never());
             Assert.Null(result.Messages);
 
             Assert.Equal(1, completeParams.BatchSummaries.Length);
             Assert.Empty(completeParams.BatchSummaries[0].ResultSetSummaries);
             Assert.NotEmpty(completeParams.BatchSummaries[0].Messages);
 
+            // ... Batch start summary should not contain result sets, messages, but should contain owner URI
+            Assert.NotNull(batchStartParams);
```

Comment:
I agree. It mostly has to do with the weirdness of having a callback that sets a variable. I'll see if I can improve upon it in the future, but for now it's a bit of a large change to make.